api/insight/socket.io.go

```diff
@@ -30,6 +30,16 @@ type SocketServer struct {
 	socketio.Server
 	params           *chaincfg.Params
 	watchedAddresses roomSubscriptionCounter
+	txGetter         txhelpers.RawTransactionGetter
+	newTxChan        chan *NewTx
+}
+
+// InsightSocketVin represents a single vin for the Insight "vin" JSON object
+// that appears in a "tx" message from the "inv" room.
+type InsightSocketVin struct {
+	TxID      string   `json:"txid,omitempty"`
+	Vout      uint32   `json:"vout,omitempty"`
+	Addresses []string `json:"addresses,omitempty"`
```

Comment:
Each field must have `omitempty` because an input consuming a previous out with an invalid/all-zero hash should be marshalled like `{}`